x-pack/legacy/plugins/actions/server/builtin_action_types/webhook.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+import { nullableType } from './lib/nullable';
+import { ActionType, ActionTypeExecutorOptions, ActionTypeExecutorResult } from '../types';
+import { assertValidInterval } from '../../../task_manager/lib/intervals';
+
+// Generic Schemas
+const PORT_MAX = 256 * 256 - 1;
+const PortSchema = schema.number({ min: 1, max: PORT_MAX });
+
+// config definition
+export type ActionTypeConfigType = TypeOf<typeof ConfigSchema>;
+
+const HeadersSchema = schema.recordOf(schema.string(), schema.string());
+const ProxySchema = schema.object({
+  host: schema.string(),
+  port: nullableType(PortSchema),
+});
+
+const ConfigSchema = schema.object(
+  {
+    host: schema.string(),
+    port: PortSchema,
+    scheme: schema.oneOf([schema.literal('http'), schema.literal('https')], {
+      defaultValue: 'http',
+    }),
+    method: schema.oneOf(
+      [
+        schema.literal('head'),
```

Comment:
`head` seems pretty weird here, I guess maybe for testing or something?  Likewise `delete`, and even `get` - I typically think of webhook as "url and request body that does something interesting", but obviously could be viewed more generally.  Thinking we might just start with `post` and `put`, expand to other methods if there's an actual requirement for it - will at least cut down on testing :-) . Also, default to `post` which is what I think I usually see.

Reply:
Reduces to POST & PUT.
I considered keeping GET but a quick 20 minutes search found very very few webhooks implementing GET, so I think it's best to remove it for now, as it's very easy to bring back later if we need it.